src/components/Input/docs/Input.stories.mdx

```diff
@@ -0,0 +1,168 @@
+import { InputPropsTable } from './InputPropsTable';
+
+<Meta
+    title="Components/Input"
+    component={Input}
+    argTypes={{
+        variant: {
+            options: ['boxed', 'bottom-lined'],
+            defaultValue: 'boxed',
+            control: 'select'
+        },
+        size: {
+            options: ['small', 'medium'],
+            defaultValue: 'medium',
+            control: 'select'
+        }
+    }}
+/>
+
+export const Template = args => <Input {...args} />;
+
+# Input
+
+Inputs are mainly used in forms to introduce alphanumeric content and edit it.
+
+<Canvas>
+    <Story
+        name="Basic"
+        args={{
+            variant: 'boxed',
+            placeholder: 'user@free-now.com',
+            label: 'E-Mail Address',
+            id: 'email'
+        }}
+    >
+        {Template.bind({})}
+    </Story>
+</Canvas>
+
+## Properties
+
+<ArgsTable of={Input} />
+
+<StyledSystemLinks component="Input" supportedProps={['margin', 'width']} />
+
+Additionally, all native HTML input element props are supported and forwarded to the underlying input element.
+
+## Controlled vs Uncontrolled
+
+<Canvas>
+    <Story name="Controlled Input">
+        {() => {
+            const [value, setValue] = useState('');
+            const handleChange = event => setValue(event.target.value);
+            return (
+                <Input
+                    value={value}
+                    onChange={handleChange}
+                    variant="boxed"
+                    placeholder="Type..."
+                    label="Controlled input"
+                    id="email"
+                />
+            );
+        }}
+    </Story>
+    <Story name="Uncontrolled Input" args={{ label: 'Uncontrolled input' }}>
+        {Template.bind({})}
+    </Story>
+</Canvas>
+
+## Testing
+
+```ts
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as React from 'react';
+import { Input } from './Input';
+
+test('allows to be tested using accessible queries', () => {
+    const spySubmit = jest.fn();
+    const onSubmit = e => {
+        e.preventDefault();
+        const formData = new FormData(e.currentTarget);
+        spySubmit({ username: formData.get('username-name'), password: formData.get('password-name') });
+    };
+
+    render(
+        <form onSubmit={onSubmit}>
+            <Input id="username-id" name="username-name" label="username" />
+            <Input id="password-id" type="password" name="password-name" label="password" />
+            <button type="submit">Submit</button>
+        </form>
+    );
+
+    const usernameInput = screen.getByRole('textbox', { name: 'username' });
+    const passwordInput = screen.getByLabelText(/password/i);
+    const submitBtn = screen.getByRole('button', { name: /submit/i });
+
+    userEvent.type(usernameInput, 'jaimito');
+    userEvent.type(passwordInput, 'tontoelquelolea');
```

Comment:
I know you didn't introduce this now, but maybe it would be nice to change this? 🤣 🤣